From 88ca55a91033cdf2bd060e01f9971f05ad033810 Mon Sep 17 00:00:00 2001
From: Herbert Xu <herbert@gondor.apana.org.au>
Date: Mon, 1 Sep 2014 13:10:36 -0400
Subject: [crypto] drbg: Fix format string for debugging statements

Message-id: <E1XORNU-00087p-MC@gondolin.me.apana.org.au>
Patchwork-id: 90067
O-Subject: [PATCH 12/19] crypto: drbg - Fix format string for debugging statements
Bugzilla: 1011562
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Rafael Aquini <aquini@redhat.com>

RHEL7 Bugzilla #1011562

Upstream commit: a9089571f2fc203c3ba6595a60f0045f048494da

The initial format strings caused warnings on several architectures. The
updated format strings now match the variable types.

Reported-by: kbuild test robot <fengguang.wu@intel.com>
Reported-by: Randy Dunlap <rdunlap@infradead.org>
CC: Joe Perches <joe@perches.com>
Signed-off-by: Stephan Mueller <smueller@chronox.de>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/crypto/drbg.c b/crypto/drbg.c
index cce915b..c9b4c49 100644
--- a/crypto/drbg.c
+++ b/crypto/drbg.c
@@ -1106,7 +1106,7 @@ static int drbg_seed(struct drbg_state *drbg, struct drbg_string *pers,
 
  /* 9.1 / 9.2 / 9.3.1 step 3 */
  if (pers && pers->len > (drbg_max_addtl(drbg))) {
-  pr_devel("DRBG: personalization string too long %lu\n",
+  pr_devel("DRBG: personalization string too long %zu\n",
     pers->len);
   return -EINVAL;
  }
@@ -1984,7 +1984,7 @@ static int __init drbg_init(void)
 
  if (ARRAY_SIZE(drbg_cores) * 2 > ARRAY_SIZE(drbg_algs)) {
   pr_info("DRBG: Cannot register all DRBG types"
-   "(slots needed: %lu, slots available: %lu)\n",
+   "(slots needed: %zu, slots available: %zu)\n",
    ARRAY_SIZE(drbg_cores) * 2, ARRAY_SIZE(drbg_algs));
   return ret;
  }
-- 
1.7.1